src/modules/ShopgateExecuteAppRelatedCode.js

```diff
@@ -0,0 +1,34 @@
+import { isShopgateIosApp } from './IsShopgateIosApp';
+import { shopgateExecuteWithRetry } from './ShopgateExecuteWithRetry';
+import { isShopgateApp } from './IsShopgateApp';
+
+/**
+ * Executes Shopgate App related Code
+ *
+ * @param {function} executeCallback callback has to return true
+ */
+export function shopgateExecuteAppRelatedCode(executeCallback) {
+  if (!window.hasOwnProperty('isShopgateApp')) {
+    if (isShopgateIosApp() && !isShopgateApp()) {
```

Comment:
Having the naming this way makes this expression contradictory. I know it is taken from the legacy implementation - what do you think about(for example):
* calling the local variable shopgateApp shopgateAppReady or isShopgateAppReady
* calling isShopgateApp isShopgateJSBridgeAvailable

Reply:
renamed the variable to shopgateApp and the method to isShopgateJSBridgeAvailable